testing/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/configurationcache/ConfigurationCacheProblemsExecutionResultFixture.groovy

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.fixtures.configurationcache
+
+import junit.framework.AssertionFailedError
+import org.gradle.integtests.fixtures.executer.ExecutionFailure
+import org.gradle.integtests.fixtures.executer.ExecutionResult
+import org.gradle.util.internal.ConfigureUtil
+import org.hamcrest.Matcher
+
+import static org.junit.Assert.assertThrows
+
+/**
+ * A `ConfigurationCacheProblemsFixture` that supports `ExecutionResult`.
+ */
+final class ConfigurationCacheProblemsExecutionResultFixture extends ConfigurationCacheProblemsFixture {
+
+    ConfigurationCacheProblemsExecutionResultFixture(File rootDir) {
+        super(rootDir)
+    }
+
+    void assertFailureHasError(
+        ExecutionFailure failure,
+        String error,
+        @DelegatesTo(value = HasConfigurationCacheErrorSpec, strategy = Closure.DELEGATE_FIRST) Closure<?> specClosure
+    ) {
+        assertFailureHasError(failure, newErrorSpec(error, ConfigureUtil.configureUsing(specClosure)))
+    }
+
+    void assertFailureHasError(
```

Comment:
🤔  If I didn't miss anything, there are no real usages of `assertFailureHasError` and its overloads on `ConfigurationCacheProblemsExecutionResultFixture`, and also no real usages of `newErrorSpec` and its overloads on `ConfigurationCacheProblemsFixture`.

If that is indeed the case, should we remove the methods altogether?

Reply:
They were unused before this change set, including `HasConfigurationCacheErrorSpec`. It seems these were used originally for asserting against the "Configuration cache state could not be cached: <reason>" CC store/load error message but no call sites remain. I may look into deleting them in a follow-up PR.